CMakeLists.txt

```diff
@@ -5,62 +5,94 @@ project(ni_grpc_device_server
   VERSION 2.7.0)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
-include(CreateVirtualEnvironment)
+
 
 # Workaround for: https://bugs.chromium.org/p/boringssl/issues/detail?id=423
 if (CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
   set(CMAKE_SYSTEM_PROCESSOR "amd64")
 endif()
 
+
+#---
+# Configuration options
+#---
+option(USE_NILRT_LEGACY_TOOLCHAIN "Enable to use tools and libraries from a NILRT compile toolchain." OFF)
+option(USE_PYTHON_VIRTUALENV "Enable to use the automatically-generated python venv, local to this project source." ON)
+option(USE_SUBMODULE_LIBS "Enable to link against the submodule libraries, rather than their native-OS equivalents." ON)
+
 #----------------------------------------------------------------------
-# Use the grpc targets directly from this build, only when not cross-compiling.
+# Setup build dependencies, according to the toolchain options.
 #----------------------------------------------------------------------
-if(CMAKE_CROSSCOMPILING)
+
+if(USE_SUBMODULE_LIBS)
+  # The archetypical WIN32 build case
+  # protobuf_INSTALL must be turned OFF whenever building it as a cmake subdir.
+  set(protobuf_INSTALL OFF)
```

Comment:
This may actually be a larger testing decision. Is it intentional that the CI pipeline is configured to use the submodule deps during a build? Normally, projects that want to support linux distributions do so by using their native toolchains to build the project. Knowing that ubuntu can build grpc-device doesn't help much if it immediately fails on install because the dynamically linked grpc package in the distro is incompatible with the submodule version used to build it.